src/test/ui/typeck/typeck-default-trait-impl-negation-sync.stderr

```diff
@@ -3,6 +3,8 @@ error[E0277]: `MyNotSync` cannot be shared between threads safely
    |
 LL | fn is_sync<T: Sync>() {}
    |    -------    ---- required by this bound in `is_sync`
+   |    |
+   |    required by a bound in `is_sync`
```

Comment:
I'm a bit confused by these diffs. That is -- I'm not quite sure what this label is trying to tell me. I guess what was happening before is that we had a span for `is_sync` but lacked more precise information, so we got no label, but now we get a label?

Nonetheless, I'm not sure this is a net *improvement* -- the label that tells you *which bound* definitely seems good, and this new label kind of distracts from that.

Would it be possible to get more precise information to help us in tracking *which label* is causing the problem?

Reply:
This was born out of https://github.com/rust-lang/rust/pull/69745#discussion_r388690977, where the small separation between the two spans made @eddyb confused for a short moment. I thought that adding some label to the span might make it clearer in all cases that we're pointing at two different things. Another alternative would be to only point at the `fn` ident if it is in a different line to the bound.

Reply:
BTW, that change is in its own commit, so you can look at the diff without it.

Reply:
I have changed the second commit to check if both spans are in the same line, and if they are, to not display the span pointing at `is_sync`.